src/app/routes/enrolment/fields/fields.component.ts

```diff
@@ -16,19 +17,30 @@ import { IFieldDefinition } from '@energyweb/credential-governance/dist/src/type
 })
 export class FieldsComponent {
   @Input() title: string;
-  @Input() isIssuerRequestView = false;
+  @Input() showMergeButton = false;
   @Input() fieldsList: KeyValue<string, string | number>[];
   @Output() fieldToCopy = new EventEmitter<KeyValue<string, string | number>>();
   @Input() issuerFields: IFieldDefinition[] | undefined;
   hasValidValue(val: string | number) {
     return val && val !== 'null';
   }
 
-  showMergeButton(field): boolean {
-    return (
-      this.isIssuerRequestView &&
-      this.hasValidValue(field?.value) &&
-      !!this.issuerFields.find((fld) => fld.label === field.label)
-    );
+  isJsonString(value: string | number) {
+    if (typeof value === 'number') {
+      return false;
+    }
+    try {
+      JSON.parse(value);
+    } catch {
+      return false;
+    }
+    return true;
+  }
+
+  parseToJson(value: string | number): JsonObject {
```

Comment:
I think this is the correct return type because the return will be undefined if `typeof value === 'number'`
```suggestion
  parseToJson(value: string | number): JsonObject | undefined {
```

Reply:
Yeah, i added this, but it not possible to happen, because `isJsonString` won't let it in there. We have first check isJsonString and later i'm parsing it. While parsing i'm 100% sure it is a string, but typings complain, that's why i added this check.

Reply:
Ok, no problem